RevenueCatUI/Data/TestData.swift

```diff
@@ -164,10 +164,10 @@ internal enum TestData {
 extension TrialOrIntroEligibilityChecker {
 
     /// Creates a mock `TrialOrIntroEligibilityChecker` with a constant result.
-    static func producing(eligibility: IntroEligibilityStatus) -> Self {
+    static func producing(eligibility: @autoclosure @escaping () -> IntroEligibilityStatus) -> Self {
```

Comment:
This allows providing a different value for each call, like using `Bool.random()`, so it's not computed eagerly.